compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java

```diff
@@ -257,7 +256,15 @@ public void visit(BLangPackage pkgNode) {
         this.currPkgEnv = pkgEnv;
         this.env = pkgEnv;
 
-        pkgNode.topLevelNodes.forEach(e -> ((BLangNode) e).accept(this));
+        for (TopLevelNode e : pkgNode.topLevelNodes) {
+            try {
+                ((BLangNode) e).accept(this);
+            } catch (InvokableAnalysisBlockedOnDependencyException ignore) {
```

Comment:
Handling Java exceptions is an expensive operation. That will slow down the analysis process. Hence shall we revert exception based logic, and improve the old status based implementation.